presto-hive/src/main/java/io/prestosql/plugin/hive/HiveConfig.java

```diff
@@ -64,6 +64,7 @@
     private int maxPartitionBatchSize = 100;
     private int maxInitialSplits = 200;
     private int splitLoaderConcurrency = 4;
+    private int maxSplitsPerSecond = Integer.MAX_VALUE;
```

Comment:
Using a huge value here leaks an implementation detail to users, since the config system will print the default values on startup. It's better to make this a nullable `Integer` with a null default value, then convert in the code where we read the config
```java
firstNonNull(hiveConfig.getMaxSplitsPerSecond(), Integer.MAX_VALUE)

Reply:
done